src/test/java/uk/gov/companieshouse/api/testdata/service/impl/AcspProfileServiceImplTest.java

```diff
@@ -83,6 +88,140 @@ void createAcspProfileWithDefaultValues() throws DataException {
         assertEquals(0L, captured.getVersion());
     }
 
+    @Test
+    void createAcspProfileWithAmlDetails() throws DataException {
+        AcspProfileSpec spec = getAcspProfileSpec();
+
+        when(randomService.getString(8)).thenReturn("randomId");
+        AcspProfile savedProfile = getAcspProfile(spec);
+
+        when(repository.save(any(AcspProfile.class))).thenReturn(savedProfile);
+
+        AcspProfileData result = service.create(spec);
+
+        assertNotNull(result);
+        assertEquals("randomId", result.getAcspNumber());
+
+        ArgumentCaptor<AcspProfile> captor = ArgumentCaptor.forClass(AcspProfile.class);
+        verify(repository).save(captor.capture());
+
+        AcspProfile captured = captor.getValue();
+        assertEquals("randomId", captured.getId());
+        assertEquals("randomId", captured.getAcspNumber());
+        assertEquals("active", captured.getStatus());
+        assertEquals("ltd", captured.getType());
```

Comment:
Maybe check these against `spec` instead of a hardcoded value
```suggestion
        assertEquals(spec.getStatus(), captured.getStatus());
        assertEquals(spec.getType(), captured.getType());
```